pkg/internal/kube/store.go

```diff
@@ -36,10 +36,27 @@ func qName(om *informer.ObjectMeta) qualifiedName {
 	return qualifiedName{name: om.Name, namespace: om.Namespace, kind: om.Kind}
 }
 
-// MetaSourceLabels allow overriding some metadata from kubernetes labels
-type MetaSourceLabels struct {
-	ServiceName      string `yaml:"service_name" env:"BEYLA_KUBE_META_SOURCE_LABEL_SERVICE_NAME"`
-	ServiceNamespace string `yaml:"service_namespace" env:"BEYLA_KUBE_META_SOURCE_LABEL_SERVICE_NAMESPACE"`
+// MetadataSources allow overriding some metadata from kubernetes labels and annotations
+type MetadataSources struct {
+	ServiceNameAnnotations      []string `yaml:"service_name_annotations" env:"BEYLA_KUBE_ANNOTATIONS_SERVICE_NAME" envSeparator:","`
+	ServiceNamespaceAnnotations []string `yaml:"service_namespace_annotations" env:"BEYLA_KUBE_ANNOTATIONS_SERVICE_NAMESPACE" envSeparator:","`
+	ServiceNameLabels           []string `yaml:"service_name_labels" env:"BEYLA_KUBE_LABELS_SERVICE_NAME" envSeparator:","`
```

Comment:
version is missing: https://github.com/open-telemetry/opentelemetry-operator/blob/aa3e781a95def24e52c0dceaf7c33ec5b6193a7e/pkg/constants/env.go#L39

you could make this even more generic:

```yaml
  meta_naming_sources:
    labels:
      service.name: ["app.kubernetes.io/name"]
      service.namespace: ["app.kubernetes.io/part-of"]
```

Reply:
Thanks, I like your proposal. Will implement it